athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/AwsRestHighLevelClientFactory.java

```diff
@@ -0,0 +1,182 @@
+    {
+        this.useDefaultCredentials = builder.useDefaultCredentials;
+        this.username = builder.username;
+        this.password = builder.password;
+    }
+
+    /**
+     * Gets an Elasticsearch REST client injected with credentials.
+     * @param endpoint is the Elasticsearch cluster's domain endpoint
+     *                 (e.g. search-movies-ne3fcqzfipy6jcrew2wca6kyqu.us-east-1.es.amazonaws.com).
+     * @return an Elasticsearch REST client. If useDefaultCredentials = true, the client is injected
+     *          with AWS credentials. If useDefaultCredentials = false and username/password are not
+     *          empty, it is injected with username/password credentials. Otherwise a default client
+     *          with no credentials is returned.
+     */
+    public AwsRestHighLevelClient getClient(String endpoint)
+    {
+        logger.info("getClient - enter: " + endpoint);
+
+        if (useDefaultCredentials) {
+            logger.info("Client injected with AWS credentials");
+
+            return new AwsRestHighLevelClient.Builder(endpoint)
+                    .withCredentials(new DefaultAWSCredentialsProviderChain()).build();
+        }
+        else if (!username.isEmpty() && !password.isEmpty()) {
+            logger.info("Client injected with username/password credentials");
+
+            return new AwsRestHighLevelClient.Builder(endpoint)
+                    .withCredentials(username, password).build();
+        }
+
+        // Default client.
+        return new AwsRestHighLevelClient.Builder(endpoint).build();
+    }
+
+    /**
+     * Gets a default client factory that will create Elasticsearch REST clients injected with AWS Credentials.
+     * @return a new default client factory.
+     */
+    public static AwsRestHighLevelClientFactory defaultFactory()
+    {
+        return new Builder().build();
+    }
+
+    /**
+     * A builder for the AwsRestHighLevelClientFactory class.
+     */
+    public static class Builder
+    {
+        private boolean useDefaultCredentials;
+        private String username;
+        private String password;
+
+        /**
+         * A constructor for the builder. As a default behaviour, it sets the client factory to use AWS credentials
+         * when constructing the client.
+         */
+        public Builder()
+        {
+            useDefaultCredentials = true;
+            username = "";
+            password = "";
+        }
+
+        /**
+         * Sets the client factory to use username/password credentials from Amazon Secrets Manager when constructing
+         * the client.
+         * @param secretName is the name of the secret.
+         * @return this.
+         */
+        public Builder withSecretCredentials(String secretName)
+        {
+            logger.info("withSecretCredentials - enter: " + secretName);
+
+            if (secretName.isEmpty()) {
+                logger.warn("Secret Name is empty.");
```

Comment:
is this really a warning? or should this throw?

Reply:
Gone due to refactoring.